widgetssdk/src/main/java/com/glia/widgets/survey/SurveyController.java

```diff
@@ -39,6 +44,22 @@ public void init(@Nullable Survey survey) {
         setQuestions(survey);
     }
 
+    private boolean isAlreadyInit(@Nullable Survey survey) {
+        if (this.survey == null) {
+            return false;
+        }
+        if (survey == null) {
+            return false;
+        }
+        if (!this.survey.getId().equals(survey.getId())) {
+            return false;
+        }
+        if (!this.survey.getEngagementId().equals(survey.getEngagementId())) {
+            return false;
+        }
+        return state.questions == null || !state.questions.isEmpty();
```

Comment:
Sorry, but I didn't got this condition. It seems to me these two parts are controversial:
If `state.questions` does not exist (null) => already initialized (I would expect that survey is not initialized in this case).
If `state.questions` not empty (there are some questions) there => already initialized (agree)

Reply:
Added tests for the `isAlreadyInit`

Reply:
Cool 👍